### ISSUE TITLE
[MacCatalyst] Poor scroll performance in MAUI ListView

### ISSUE THREAD

**Issue by ashok-kuvaraja** [dotnet/maui#12388]
### Description

I have a simple listview sample and load Grid with some labels as an item template to the listview. I have faced a poor scrolling performance in the MAUI ListView when scrolling it in the initial loading. First time when an item adding in the ListView, it affects the scrolling performance. Please check the behavior by the provided sample and do the needful.

### Steps to Reproduce

1. Run the provided app
2. Scroll the list view in vertical direction.
[SimpleMauiApp.zip](https://github.com/dotnet/maui/files/10335725/SimpleMauiApp.zip)

### Link to public reproduction project repository

https://github.com/ashok-kuvaraja/ListViewMauiSample

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS, I was *not* able test on other platforms

### Affected platform versions

Mac 13.1

**Comment by PureWeen** [comment]
@ashok-kuvaraja can you upload your reproduction to a github repository so it's easier to view/comment on?

**Comment by ashok-kuvaraja** [comment]
@PureWeen - I have provided the GitHub repository for the issue reproduction sample. Please check now.

**Comment by marcvanbreemen** [comment]
You're using a Nuget package from Syncfusion for the datagrid, so it's not directly a MAUI problem, unless the poor performance is also there with the MAUI's DataGrid control.

Have you reported this to Syncfusion?

**Comment by drasticactions** [comment]
> You're using a Nuget package from Syncfusion for the datagrid, so it's not directly a MAUI problem, unless the poor performance is also there with the MAUI's DataGrid control.
> 
> Have you reported this to Syncfusion?

They're not, the linked GitHub is a standard ListView with a list of items.

Some notes from what I've seen so far:

1. This is not just Catalyst, it also happens on iOS.
2. Not related to Hot Reload. If you turn it off, it still scroll bad.
3. Not related to the StackView. If you replace it with a Grid, it would do the same thing.
4. If you scroll enough, you'll start seeing dropped items not being rendered. Inspecting it, some cells are not being rendered.

<img width="382" alt="スクリーンショット 2023-02-06 20 10 46" src="https://user-images.githubusercontent.com/898335/216959130-d1aa53f8-b77b-4943-8767-1bcd05faea8e.png">

<img width="1512" alt="スクリーンショット 2023-02-06 20 14 30" src="https://user-images.githubusercontent.com/898335/216959220-b1f9b15c-7763-4660-baca-4da0f76139ac.png">

5. If you just have a label in the ViewCell, with no set height, it will render with a height of zero. When placed in a Grid, it auto centers the cell to the middle and will set a height automatically. If you use a stack layout, it will render the cells left to right.

TL;DR there's quite a few issues.

**Comment by ashok-kuvaraja** [comment]
@drasticactions - Is there any further update on this?

**Comment by drasticactions** [comment]
@ashok-kuvaraja No

**Comment by jonathanpeppers** [comment]
Other examples of a similar issue:
* https://github.com/dotnet/maui/issues/14227
* https://github.com/BECInc/MauiSpeedTest
* https://github.com/BECInc/XamarinComparison

**Comment by KarthikRajaKalaimani** [comment]
My customers are facing the same issue. When can i expect smooth scrolling ?

**Comment by jonathanpeppers** [comment]
@KarthikRajaKalaimani maybe you can share a `.speedscope` file of a Release build of your app in action?

https://aka.ms/profile-maui

That would help us find problem areas to improve.

**Comment by jinxinjuan** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 2.0. Can repro on iOS platform with sample project.
[ListViewMauiSample](https://github.com/ashok-kuvaraja/ListViewMauiSample)

**Comment by jonathanpeppers** [comment]
@jinxinjuan were you able to record a `.speedscope` file?

**Comment by RasikaPalanisamy** [comment]
@jonathanpeppers, could you kindly provide an explanation of your testing approach utilizing the `.speedscope` file provided? Additionally, could you share any guidelines or instructions to enable us to perform the same testing on our end?

**Comment by jonathanpeppers** [comment]
https://speedscope.app/ is where you would go to open this file.

https://aka.ms/profile-maui should have some examples and links inside.

**Comment by ashok-kuvaraja** [comment]
Hi @jonathanpeppers,

May I know the status of this issue and when will it be resolved? Have you checked the performance issue with the given [speedscope](https://github.com/dotnet/maui/files/12180749/SpeedScope.zip) file?

Regards,
Ashok K

**Comment by jonathanpeppers** [comment]
@ashok-kuvaraja the trace looks like it is a .NET 7 app. Have you noticed any improvement in .NET 8 RC 1?

When testing performance, are you testing a `Release` build on a physical device?

**Comment by ashok-kuvaraja** [comment]
Hi @jonathanpeppers,

The reported issue persists in .NET 8 RC1 also in both release and debug mode.

**Comment by jonathanpeppers** [comment]
Do you have an updated `.speedscope` file for a `Release` build on .NET 8?

**Comment by ot-ashish-jain** [comment]
Any updates on this?

**Comment by PureWeen** [comment]
> Do you have an updated `.speedscope` file for a `Release` build on .NET 8?
